monitoring/reporter/expvar/expvars.go

```diff
@@ -0,0 +1,70 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package expvar
+
+import (
+	"expvar"
+	"net/http"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/mapstr"
+	"github.com/elastic/elastic-agent-libs/transform/typeconv"
+	"github.com/elastic/elastic-agent-shipper/monitoring/reporter"
+)
+
+//ExpvarsConfig is the config struct for marshalling whatever we get from the config file
+type ExpvarsConfig struct {
+	Enabled bool   `config:"enabled"`
+	Addr    string `config:"address"`
+	Name    string `config:"name"`
+}
+
+// Expvars is the simple manager for the expvars web interface
+type Expvars struct {
+	log     *logp.Logger
+	metrics reporter.QueueMetrics
+}
+
+// NewExpvarReporter initializes the expvar interface, and starts the http frontend.
+func NewExpvarReporter(cfg ExpvarsConfig) reporter.Reporter {
+	exp := Expvars{
+		log: logp.L(),
+
+		metrics: reporter.QueueMetrics{},
+	}
+	exp.log.Debugf("Starting expvar monitoring...")
+	expvar.Publish(cfg.Name, expvar.Func(exp.format))
+	exp.runFrontend(cfg)
+	return &exp
+}
+
+func (exp Expvars) runFrontend(cfg ExpvarsConfig) {
+	go func() {
+		err := http.ListenAndServe(cfg.Addr, nil)
```

Comment:
It looks like we need a way to trigger the HTTP server to stop and force the goroutine to exit.

Reply:
Agreed, starting to think the sigint handling stuff should just be crammed into this PR.